### PR TITLE
Restore form tests

### PR DESCRIPTION
**Pull request by denniskigen**
Refactors and restores outdated tests for the following components:

- `ConditionsForm`
- `ProgramsForm`
- `VisitNotesForm`

Also includes small refactors to overview components and app entry points, mostly involving renaming extension slot
names to include the word `form`. This hopefully helps to disambiguate extension slots for forms. 

Overall, coverage for these tests exceeds 80%.

This PR is still a draft because I haven't finished working out a small problem with the `VisitNotesForm` component test.

### CODE REVIEW

File: packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { SyntheticEvent } from 'react';
 import dayjs from 'dayjs';
-import debounce from 'lodash-es/debounce';
+import { debounce } from 'lodash-es';
```
Comment (FlorianRappl) [inline packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx:3]: Why this change? Please keep the submodule imports. Otherwise you have the full lodash in here.

File: packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx [hunk 2]
```diff
@@ -319,7 +320,7 @@ const ConditionsForm: React.FC<ConditionsFormProps> = ({ patientUuid, isTablet }
         </DatePicker>
       )}
       <div style={{ marginTop: '1.625rem' }}>
-        <Button style={{ width: '50%' }} kind="secondary" type="button" onClick={closeWorkspace}>
+        <Button style={{ width: '50%' }} kind="secondary" type="button" onClick={() => closeWorkspace()}>
```
Comment (FlorianRappl) [inline packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx:323]: Why the extra function?

Comment (denniskigen) [reply packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx]: Fixed in the latest commit.

File: packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx (hunk 1 above)
Comment (denniskigen) [reply packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx]: Fixed in the latest commit.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks! :beers:
